src/EmotiBitFactoryTest.cpp

```diff
@@ -34,32 +34,54 @@ const char* EmotiBitFactoryTest::TypeTag::LED_YELLOW_OFF = "-Y\0";
 const char* EmotiBitFactoryTest::TypeTag::NULL_VAL = "\0";
 const char* EmotiBitFactoryTest::TypeTag::TEST_PASS = "PASS\0";
 const char* EmotiBitFactoryTest::TypeTag::TEST_FAIL = "FAIL\0";
+const char* EmotiBitFactoryTest::TypeTag::VERSION_VALIDATION = "VV\0";
+const char* EmotiBitFactoryTest::TypeTag::SKU_VALIDATION = "SV\0";
 
-void EmotiBitFactoryTest::updateOutputString(char* output, const char* testType, const char* result)
+#ifdef ARDUINO
+void EmotiBitFactoryTest::updateOutputString(String &output, const char* testType, const char* result)
 {
-	//char output[10]; // max posible for any test
-	strcat(output,testType);
-	strcat(output,TypeTag::TEST_RESULT_DELIMITER);
-	strcat(output,result);
-	if(result != TypeTag::TEST_FAIL && result != TypeTag::NULL_VAL)
+	output += testType;
+	output += TypeTag::TEST_RESULT_DELIMITER;
+	output += result;
+	if(result != TypeTag::NULL_VAL)
 	{
-		strcat(output,TypeTag::TEST_TYPE_DELIMITER);
+		output += TypeTag::TEST_TYPE_DELIMITER;
 	}
 	else
 	{
-		strcat(output, "***");
+		output += MSG_TERM_CHAR;
 	}
 }
-#ifdef ARDUINO
-	void EmotiBitFactoryTest::sendMessage(String typeTag, String payload)
+
+void EmotiBitFactoryTest::sendMessage(String typeTag, String payload)
+{
+	Serial.print(EmotiBitFactoryTest::MSG_START_CHAR);
+	Serial.print(typeTag);
+	if (!payload.equals(""))
+	{
+		Serial.print(EmotiBitFactoryTest::PAYLOAD_DELIMITER);
+		Serial.print(payload);
+	}
+	Serial.println(EmotiBitFactoryTest::MSG_TERM_CHAR);
+}
+
+	// returns the version from barcode string "SKU-VERSION-NUMBER"
+void EmotiBitFactoryTest::parseBarcode(Barcode* barcode)
+{
+	String tempEmotibitVersion;
+	barcode->sku = barcode->code.substring(0, barcode->code.indexOf(BARCODE_DELIMITER));
+	barcode->code = barcode->code.substring(barcode->code.indexOf(BARCODE_DELIMITER) + 1);
+	tempEmotibitVersion = barcode->code.substring(0, barcode->code.indexOf(BARCODE_DELIMITER));
+	barcode->emotibitNumber = barcode->code.substring(barcode->code.indexOf(BARCODE_DELIMITER) + 1);
+	// if barcode has V4
+	if (tempEmotibitVersion.equals("V4"))
```

Comment:
@produceconsumerobot we need to discuss the best way to remove this.

Right now, in the FW, EmotiBitVersion::V04a != 4, its equal to 6. So converting "V4" to int and using that for version ceck might not be the best approach.